File: examples/image/image.js
let img;

function setup(c) {
  img = c.loadImage("the-great-wave-kanagawa.jpg");
  c.noLoop();
}

function draw(c) {
  c.clear();
  c.image(img, 0, 0, c.width, c.height);
  let fullCanvas = c.get(0, 0, c.width, c.height);
  c.set(c.width / 2, 0, fullCanvas);
}

File: examples/image/image.go
package main

import (
	"context"

	"github.com/emprcl/runal"
)

func main() {
	runal.Run(context.Background(), setup, draw, nil, nil)
}

var img runal.Image

func setup(c *runal.Canvas) {
	img = c.LoadImage("the-great-wave-kanagawa.jpg")
	c.NoLoop()
}

func draw(c *runal.Canvas) {
	c.Image(img, 0, 0, c.Width, c.Height)

	fullCanvas := c.Get(0, 0, c.Width, c.Height)

	c.Image(fullCanvas, c.Width/2, 0, c.Width/2, c.Height)
}
